inc/classes/subscriber/Tools/class-detect-missing-tags-subscriber.php

```diff
@@ -0,0 +1,95 @@
+<?php
+namespace WP_Rocket\Subscriber\Tools;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WP_Rocket\Logger\Logger;
+
+/**
+ * Detect and report when <html>, wp_footer() and <body> tags are missing.
+ *
+ * @since  3.4.2
+ * @author Soponar Cristina
+ */
+class Detect_Missing_Tags_Subscriber implements Subscriber_Interface {
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'admin_notices'                      => 'notice_missing_tags',
+			'before_rocket_maybe_process_buffer' => 'maybe_missing_tags',
+		];
+	}
+
+	/**
+	 * Check if there is a missing </html> or </body> tag
+	 *
+	 * @since  3.4.2
+	 * @author Soponar Cristina
+	 *
+	 * @param string $html HTML content.
+	 */
+	public function maybe_missing_tags( $html ) {
+		Logger::info( 'START Detect_Missing_Tags_Subscriber - maybe_missing_tags ', [ 'maybe_missing_tags' ] );
+
+		// Remove all comments before testing tags. If </html> or </body> tags are commented this will identify it as a missing tag.
+		$html    = preg_replace( '/<!--([\\s\\S]*?)-->/', '', $html );
+		$is_html = true;
+		if ( ! preg_match( '/(<\/html>)/i', $html ) ) {
+			$is_html = false;
+			Logger::debug( 'Not found closing </html> tag.', [ 'maybe_missing_tags' ] );
+		}
+
+		$is_body = true;
+		if ( ! preg_match( '/(<\/body>)/i', $html ) ) {
+			$is_body = false;
+			Logger::debug( 'Not found closing </body> tag.', [ 'maybe_missing_tags' ] );
+		}
+
+		$did_wp_footer = true;
+		if ( did_action( 'wp_footer' ) === 0 ) {
+			$did_wp_footer = false;
+			Logger::debug( 'Did action did not run wp_footer() function.', [ 'maybe_missing_tags' ] );
+		}
+
+		set_transient( 'rocket_missing_tags', $is_html && $is_body && $did_wp_footer, HOUR_IN_SECONDS );
+	}
+
+	/**
+	 * This notice is displayed if there is a missing required tag or function: </html>, </body> or wp_footer()
+	 *
+	 * @since  3.4.2
+	 * @author Soponar Cristina
+	 */
+	public function notice_missing_tags() {
+		$screen = get_current_screen();
```

Comment:
This notice should appear only in WP Rocket Settings page?

Reply:
yes